src/librustc_metadata/creader.rs

```diff
@@ -853,6 +853,33 @@ impl<'a> CrateLoader<'a> {
         }
     }
 
+    fn inject_profiler_runtime(&mut self) {
+        if self.sess.opts.debugging_opts.profile {
+            let mut uses_std = false;
+            self.cstore.iter_crate_data(|_, data| {
```

Comment:
@alexcrichton Sure, I'll remove it.

I imagine someone would replace the libc calls in the stock profiler builtins with something else and then link to that, not write them from scratch. I am thinking of doing this for my embedded system work, incidentally.